prime-router/src/main/kotlin/fhirengine/engine/FHIRTranslator.kt

```diff
@@ -47,83 +50,124 @@ class FHIRTranslator(
+                    return if (message.topic.isSendOriginal) {
+                        listOf(sendOriginal(message, receiver, actionHistory))
+                    } else {
+                        listOf(sendTranslated(message, receiver, actionHistory))
+                    }
                 }
-
-                return listOf(
-                    FHIREngineRunResult(
-                        event,
-                        report,
-                        blobInfo.blobUrl,
-                        queueMessage
+                else -> {
+                    throw RuntimeException(
+                        "Message was not a FhirTranslateQueueMessage and cannot be " +
+                            "processed by FHIRTranslator: $message"
                     )
-                )
-            }
-            else -> {
-                throw RuntimeException(
-                    "Message was not a FhirTranslateQueueMessage and cannot be processed by FHIRTranslator: $message"
-                )
+                }
             }
         }
     }
 
+    /**
+     * Get the greatest ancestor of the report and send the blob associated with it to the receiver. This is the
+     * "original message pass through" feature, documented here:
+     *  prime-router/docs/design/proposals/0024-original-message-passthrough.md
+     * [message] defines reportId and topic
+     * [receiver] the receiver to send the original message to
+     * [actionHistory] ensures all activities are recorded to the database
+     */
+    internal fun sendOriginal(
+        message: FhirTranslateQueueMessage,
+        receiver: Receiver,
+        actionHistory: ActionHistory,
+    ): FHIREngineRunResult {
+        logger.trace("Preparing to send original message")
+        val originalReport = reportService.getRootReport(message.reportId)
+        val bodyBytes = BlobAccess.downloadBlobAsByteArray(originalReport.bodyUrl)
+
+        // get a Report from the message
+        val (report, event, blobInfo) = Report.generateReportAndUploadBlob(
+            Event.EventAction.SEND,
+            bodyBytes,
+            listOf(message.reportId),
+            receiver,
+            this.metadata,
+            actionHistory,
+            topic = message.topic,
+            originalReport.externalName
+        )
+
+        return FHIREngineRunResult(
+            event,
+            report,
+            blobInfo.blobUrl,
+            ReportEventQueueMessage(Event.EventAction.SEND, false, report.id, OffsetDateTime.now().toString())
+        )
+    }
+
+    /**
+     * Translate the FHIR bundle associated with the report ID to the format the receiver specified and let the
+     *  batch step pick it up.
+     * [message] defines reportId, topic, and the FHIR bundle to translate
+     * [receiver] the receiver to send the translated message to
+     * [actionHistory] ensures all activities are recorded to the database
+     */
+    internal fun sendTranslated(
```

Comment:
This is the functionality of the old translate step that translated all bundles per receiver's settings.